2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,383 @@
+  },
+  "Configuration": "[1.3.1,2.0)"
+}
+```
+
+>[!NOTE]
+>Credentials management will be using an upcoming RFC for generalized
+>credential management.  Not sure how to make that work in json other
+>than requiring json to have credential as a json object but that means
+>password/secret is in clear text.
+
+The `repository` property can be the name of the registered repository or the URL
+to a repository.
+If no `repository` is specified, it will use the `default` repository.
+
+The older `System.Version` four part version type will be supported to retain
+compatibility with existing published modules using that format.
+
+Due to the differences between semver and `System.Version`, we may have to keep
+the two distinct.
+
+Dependent declared modules not found on the system will prompt to be installed
+including module name, version, and size unless `-IncludeDependencies` is
+specified which will install without prompting.
+Declared dependencies are only searched within the same repository as the original
+module to be installed.
+
+### Saving resources
+
+This cmdlet uses `Install-PSResource -DestinationPath`.
+
+With the removal of PackageManagement, there is still a need to support saving
+arbitrary nupkgs (assemblies) used for scripts.
+
+`Save-PSResource -Type Library` will download nupkgs that have a `lib` folder.
+The dependent native library in `runtimes` matching the current system runtime
+will be copied to the root of the destination specified.
+A `-IncludeAllRuntimes` can be used to explicitly retain the `runtimes` directory
+hierarchy within the nupkg to the root of the destination.
+
+>[!NOTE]
+>Library support may not be available in 3.0.
+
+A `-Prerelease` switch allows saving prerelease versions.
+If the `-Version` includes a prerelease label like `2.0.0-beta4`, then this
+switch is not necessary and the prerelease version will be installed.
+
+A `-AsNupkg` switch will save the resource as a nupkg (if it was originally a
+nupkg) instead of expanding it into a folder.
+
+### Updating resources
+
+`Update-PSResource` will update all resources to most recent minor version by default.
+A `-AllowMajorVersionUpdate` switch will allow updating to newer major version.
+A `-PatchUpdateOnly` switch will allow updating only to patched versions.
+
+If the installed resource is a pre-release, it will automatically update to
+latest prerelease or stable version (if available).
+
+### Publishing resources
+
+`Publish-PSResource` will supporting publishing modules and scripts.
+It will follow the same model as `Publish-Module` in v2.
+
+A `-DestinationPath` can be used to publish the resulting nupkg locally.
+`Publish-PSResource` can accept a path to the nupkg to publish to a repository.
+
+A `-Nuspec` parameter can be used to specify a nuspec file rather than relying
+on this module to produce one.
+
+A `-SkipDependenciesCheck` switch can be used to bypass the default check that
+all dependencies are present.
+
```

Comment:
So a `-Trusted` type of switch?

Reply:
I don't know if there should be a switch for this. If I won't be able to use resources due to policy, why even show them to me in the first place, and why let me install them? Wouldn't it be better to follow policy that may be set by my IT department, only showing me resources that I'm allowed to install when I use `Find-...`, and only allowing me to install those resources that are compliant with my policy when I use `Install-...`? Otherwise we're letting users find and install resources that they can't use.

How `RemoteSigned` plays with this is also interesting because these resources are not treated as remote when they are installed, so they bypass that policy.

If you want a switch to filter the view though, I would go with `-Signed` instead of `-Trusted`, since it's more explicit.

I'm not sure of an exact solution here (I should put more thought into it first), but it would be good to continue discussing (if you're reading this and have thoughts/ideas, please share) so that we have a plan that makes sense. Allowing people to find software (scripts or modules) and install software (scripts or modules) that they can't even use doesn't sound like the right thing to do. I also wonder if allowing people to find scripts or modules in repositories on the internet and install them without them marked as remote with an alternate file stream is the right thing to do.

Reply:
For now, I'll add it to the alternate considerations section as something to consider in the future